src/server/scripts/Outland/TempestKeep/botanica/boss_thorngrin_the_tender.cpp

```diff
@@ -17,152 +17,161 @@
+            return;
 
-            void UpdateAI(uint32 diff) override
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            switch (eventId)
             {
-                if (!UpdateVictim())
-                    return;
-
-                events.Update(diff);
-
-                if (me->HasUnitState(UNIT_STATE_CASTING))
-                    return;
-
-                while (uint32 eventId = events.ExecuteEvent())
-                {
-                    switch (eventId)
-                    {
-                        case EVENT_SACRIFICE:
-                            if (Unit* target = SelectTarget(SelectTargetMethod::Random, 1, 0.0f, true))
-                            {
-                                Talk(SAY_CAST_SACRIFICE);
-                                DoCast(target, SPELL_SACRIFICE, true);
-                            }
-                            events.ScheduleEvent(EVENT_SACRIFICE, 29400ms);
-                            break;
-                        case EVENT_HELLFIRE:
-                            Talk(SAY_CAST_HELLFIRE);
-                            DoCastVictim(SPELL_HELLFIRE, true);
-                            if (IsHeroic())
-                                events.ScheduleEvent(EVENT_HELLFIRE, 17400ms, 19300ms);
-                            else
-                                events.ScheduleEvent(EVENT_HELLFIRE, 18s);
-                            break;
-                        case EVENT_ENRAGE:
-                            Talk(EMOTE_ENRAGE);
-                            DoCast(me, SPELL_ENRAGE);
-                            events.ScheduleEvent(EVENT_ENRAGE, 33s);
-                            break;
-                        default:
-                            break;
-                    }
-
-                    if (me->HasUnitState(UNIT_STATE_CASTING))
-                        return;
-                }
-
-                DoMeleeAttackIfReady();
+                case EVENT_SACRIFICE:
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 1, 0.0f, true))
+                        DoCast(target, SPELL_SACRIFICE);
+                    events.Repeat(25s, 30s);
+                    break;
+                case EVENT_HELLFIRE:
+                    DoCastSelf(SPELL_HELLFIRE);
+                    events.Repeat(15s, 25s);
+                    break;
+                case EVENT_ENRAGE:
+                    DoCastSelf(SPELL_ENRAGE);
+                    events.Repeat(30s, 40s);
+                    break;
+                default:
+                    break;
             }
 
-        private:
-            bool _phase1;
-            bool _phase2;
-        };
-
-        CreatureAI* GetAI(Creature* creature) const override
-        {
-            return GetBotanicaAI<boss_thorngrin_the_tenderAI>(creature);
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
         }
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    bool _phase1;
```

Comment:
would merge `_phase1` and `_phase2` into an uint8 with values 1 and 2 respectively